### DISCUSSION TITLE
quart equivalent of flask.sansio.scaffold

### DISCUSSION THREAD

**Discussion by cpvandehey** [pallets/quart#393] [Q&A] [answered]
Hey Quart folks! 

Im really impressed with the repo so far :) I find it to be super easy to navigate and reason with. The docs have also been great

My main question that I'm still struggling with is how people have migrated `scaffold` usages (`flask.sansio.scaffold`) from Flask to Quart. 

At the moment, I have a project that is subclassing the scaffold class and overriding `def route`:
```
class CustomScaffold(scaffold.Scaffold):
  def route(
        self,
        rule: str,
        **options: Any,
    ) -> Callable:
        """
        Wrapper over flask.Scaffold.route to provide a nicer API.
        Args:
            - method (str): one of the HTTP methods
            - rule (str): the route URI e.g. '/hello'
        """
        ...
```
And then my blueprint uses this:
```
class APIBlueprint(flask.Blueprint, scaffold.CustomScaffold):
    def __init__(self, *args, **kwargs):
        super(APIBlueprint, self).__init__(*args, **kwargs)

    def route(self, *args, **kwargs):
        return scaffold.CustomScaffold.route(self, *args, **kwargs)
```

I use this to plug in my custom authentication logic as well as adding headers for certain situations. Before flask 2.0.0 was released, I used `RouteMixin`s to achieve this.

Im curious what the community thinks of the migration here:
* Is there a way to use `scaffold` implementations in Quart
* Am I better off trying to shove what I have in my `scaffold` implementation inside of a blueprint?

Thanks for any insights. Cheers!

**Comment by davidism** [comment, marked as answer]
You're not supposed to use scaffold at all, it's internal. Flask and Quart use it, you should be using their app and blueprint classes.
